src/shared/azureagent/resources.go

```diff
@@ -0,0 +1,70 @@
+package azureagent
+
+import (
+	"context"
+	"fmt"
+	"github.com/otterize/intents-operator/src/shared/errors"
+	"regexp"
+	"strings"
+)
+
+const StorageAccountPattern = `providers/Microsoft.Storage/storageAccounts/([^/]+)`
+
+func (a *Agent) ValidateScope(ctx context.Context, scope string) error {
+	res, err := a.resourceClient.GetByID(ctx, scope, "2022-09-01", nil)
+	if err != nil {
+		return err
+	}
+	if res.GenericResource.ID == nil {
+		return errors.Errorf("scope %s not found", scope)
+	}
+	return nil
+}
+
+func (a *Agent) GetFullResourceScope(ctx context.Context, partialScope string) (string, error) {
```

Comment:
Actually, for any resource that's not a storage account, I wouldn't go into this lookup function. All other resources AFAIK do not define any cross-subscription uniqueness, so if, for example, our customers define a partial resource group scope `/resourceGroups/myResourceGroup`, there's nothing guaranteeing uniqueness cross subscription. 
Whats more, this can lead to breaking existing behavior. So far, on partial scopes, we assumed they'd belong to the same subscription and resource group as the AKS cluster. 

Instead, let's do this:
- if the partial scope is a storage account (or a container in it), lookup for it cross-subscriptions (by storage account name, which is globally unique). 
- otherwise, fallback to assuming its in the same subscription & resource group as the AKS cluster (keeping existing behavior).